src/Compilers/CSharp/Portable/Symbols/Source/SourceAssemblySymbol.cs

```diff
@@ -679,17 +679,17 @@ private static void ReportDiagnosticsForUnsafeSynthesizedAttributes(CSharpCompil
             // As in Dev10 (see GlobalAttrBind::EmitCompilerGeneratedAttrs), we only synthesize this attribute if
             // UnverifiableCodeAttribute is found.
             Binder.ReportUseSiteDiagnosticForSynthesizedAttribute(compilation,
-                WellKnownMember.System_Security_UnverifiableCodeAttribute__ctor, diagnostics, NoLocation.Singleton);
+                WellKnownMember.System_Security_UnverifiableCodeAttribute__ctor, recordUsage: true, diagnostics, NoLocation.Singleton);
 
 
-            TypeSymbol securityPermissionAttribute = compilation.GetWellKnownType(WellKnownType.System_Security_Permissions_SecurityPermissionAttribute);
+            TypeSymbol securityPermissionAttribute = compilation.GetWellKnownType(WellKnownType.System_Security_Permissions_SecurityPermissionAttribute, recordUsage: false);
             Debug.Assert((object)securityPermissionAttribute != null, "GetWellKnownType unexpectedly returned null");
             if (securityPermissionAttribute is MissingMetadataTypeSymbol)
             {
                 return;
             }
 
-            TypeSymbol securityAction = compilation.GetWellKnownType(WellKnownType.System_Security_Permissions_SecurityAction);
+            TypeSymbol securityAction = compilation.GetWellKnownType(WellKnownType.System_Security_Permissions_SecurityAction, recordUsage: false);
```

Comment:
>compilation.GetWellKnownType(WellKnownType.System_Security_Permissions_SecurityAction, recordUsage: false) [](start = 40, length = 106)

nit: I assume that some of the types don't record usage because they are referenced by other types which did record. This may be worth a comment?

Reply:
> nit: I assume that some of the types don't record usage because they are referenced by other types which did record. This may be worth a comment?

Recorded below.

---
In reply to: [350924620](https://github.com/dotnet/roslyn/pull/40010#discussion_r350924620) [](ancestors = 350924620)